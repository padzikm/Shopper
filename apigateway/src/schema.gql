# Comments in GraphQL are defined with the hash (#) symbol.

  type Products {
    name: String
  }

  # The "Query" type is the root of all GraphQL queries.
  # (A "Mutation" type will be covered later on.)
  type Query {
    products: [Products]
  }