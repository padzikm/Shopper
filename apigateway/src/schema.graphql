type Product {
  name: String!
  desc(unit: LengthUnit): String
}

type Query {
  products: [Product!]!
}

enum LengthUnit{
  METER
  CENT
}

type Mutation{
  crProduct(unit: LengthUnit): Product
}