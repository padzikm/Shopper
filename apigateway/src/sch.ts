import {objectType, queryType, makeSchema} from 'nexus'

const products = [
    {
      name: 'raz'
    },
    {
      name: 'dwaa'
    },
    {
      name: 'trzyy'
    },
  ];

const Product = objectType({
name: 'Product',
definition(t){
    t.string('name')
}
});

const Query = queryType({
    definition(t){
        t.list.field('products', {type: Product, resolve(){ return products;}})
    }
})

export const schema = makeSchema({
    types: [Query, Product],
    outputs: {
        schema: __dirname + "/schema.graphql",
        typegen: __dirname + "/typings.ts"
    }
})

// const cokolwiek =  gql`
//   # Comments in GraphQL are defined with the hash (#) symbol.

//   type Products {
//     name: String
//   }

//   # The "Query" type is the root of all GraphQL queries.
//   # (A "Mutation" type will be covered later on.)
//   type Query {
//     products: [Products]
//   }
// `;