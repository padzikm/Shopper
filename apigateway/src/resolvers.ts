import { Query, Product, IResolvers, QueryResolvers, ProductResolvers, MutationResolvers } from "../typings/types.server";


const products = [
    {
      name: 'raz'
    },
    {
      name: 'dwaa'
    },
    {
      name: 'trzyy'
    },
  ];

const Query: QueryResolvers = {
    products: () => products
}

const Mutation: MutationResolvers = {

}

const Product: ProductResolvers = {
    name: (parent) => parent.name,
    desc: (parent, args) => args.unit ? 'jest opis' : 'nie ma opisu'
}

export const resolvers: IResolvers = {
    Query,
    Mutation,
    Product
}