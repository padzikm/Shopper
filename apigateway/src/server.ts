const { ApolloServer, gql } = require('apollo-server');

import {resolvers} from './resolvers'
import typeDefs from './schema.graphql'
//import { makeExecutableSchema } from 'graphql-tools'

const apolloResolvers = resolvers as any;

//var schema = makeExecutableSchema({typeDefs, resolvers: apolloResolvers})
// In the most basic sense, the ApolloServer can be started
// by passing type definitions (typeDefs) and the resolvers
// responsible for fetching the data for those types.
export const server = new ApolloServer({ typeDefs, resolvers });
//export const server = new ApolloServer({ schema });
//export const server = new ApolloServer(typeDefs, resolvers);

