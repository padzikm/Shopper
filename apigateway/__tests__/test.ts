import { createTestClient } from 'apollo-server-testing';
import { server } from '../src/server';
import { gql } from 'apollo-server-core';

test('create test client', () => {
    const { query, mutate } = createTestClient(server);

    expect(query).toBeDefined()
    expect(mutate).toBeDefined()
});

test('query schema test', async () => {
    const { query } = createTestClient(server);

    const queryGql = gql`query {
        __schema{
            types {
                name
            }
        }
    }`

    let res = await query({query: queryGql})
    
    expect(res.data).toMatchSnapshot()
});

test('query products test', async () => {
    const { query } = createTestClient(server);

    const queryGql = gql`query {
        products {
            name
        }
    }`

    let res = await query({query: queryGql})
    
    expect(res.data).toMatchSnapshot()
});