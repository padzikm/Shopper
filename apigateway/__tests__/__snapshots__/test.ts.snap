// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query products desc test 1`] = `
Object {
  "products": Array [
    Object {
      "desc": "nie ma opisu",
    },
    Object {
      "desc": "nie ma opisu",
    },
    Object {
      "desc": "nie ma opisu",
    },
  ],
}
`;

exports[`query products test 1`] = `
Object {
  "products": Array [
    Object {
      "name": "raz",
    },
    Object {
      "name": "dwaa",
    },
    Object {
      "name": "trzyy",
    },
  ],
}
`;

exports[`query schema test 1`] = `
Object {
  "__schema": Object {
    "types": Array [
      Object {
        "name": "Query",
      },
      Object {
        "name": "Product",
      },
      Object {
        "name": "String",
      },
      Object {
        "name": "LengthUnit",
      },
      Object {
        "name": "Mutation",
      },
      Object {
        "name": "__Schema",
      },
      Object {
        "name": "__Type",
      },
      Object {
        "name": "__TypeKind",
      },
      Object {
        "name": "Boolean",
      },
      Object {
        "name": "__Field",
      },
      Object {
        "name": "__InputValue",
      },
      Object {
        "name": "__EnumValue",
      },
      Object {
        "name": "__Directive",
      },
      Object {
        "name": "__DirectiveLocation",
      },
    ],
  },
}
`;
